File: P4/civitasruby/lib/civitas/casilla_juez.rb
#encoding:utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
  
  class Casilla_juez < Casilla
    
    def initialize(carcel,nombre)
      super(nombre)
      @carcel = carcel
    end
    
    @override
    def to_s
      "#{super}   El jugador va a la carcel"
    end
    
    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        super
        todos.at(actual).encarcelar(@carcel)
      end
    end
    
  end
  
end

File: Parcial2/Civitas/src/civitas/CasillaJuez.java
package civitas;
import java.util.ArrayList;
/**
 *
 * @author pingu
 */
public class CasillaJuez extends Casilla {
    private static int carcel;
    
    CasillaJuez(int carcel, String nombre){
        super(nombre);
        this.carcel = carcel;
    }
    
    @Override
    public String toString(){
        String info = super.toString();
        info += "\n¡Has caído en el Juez, VE A LA CÁRCEL!";
        return info;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).encarcelar(carcel);
        }
    }
    
}
